src/battle_script_commands.c

```diff
@@ -3793,6 +3793,21 @@ void SetMoveEffect(bool32 primary, bool32 certain)
                 }
                 SetMoveEffect(primary, certain);
                 break;
+            case MOVE_EFFECT_PSYCHIC_NOISE:
+                if (GetBattlerAbility(gEffectBattler) == ABILITY_AROMA_VEIL || GetBattlerAbility(BATTLE_PARTNER(gEffectBattler)) == ABILITY_AROMA_VEIL)
+                {
+                    gBattlerAbility = gEffectBattler;
+                    BattleScriptPush(gBattlescriptCurrInstr + 1);
+                    gBattlescriptCurrInstr = BattleScript_AromaVeilProtectsRet;
```

Comment:
Two things:
1 You should check ingame if there should be an Ability pop-up here. Usually when a secondary effect is blocked by an Ability (eg. Water Veil blocking burn) no Ability pop-up is shown -- the effect just doesn't happen.
2. If there _is_ supposed to be an Ability pop-up, it'll say the wrong Pokemon's Ability in the case where it's the partner's Ability that blocks it (you're setting `gBattlerAbility` to `gEffectBattler` regardless of the Ability holder)/.

Reply:
1. There is the function `IsAbilityOnSide` that is used for `ABILITY_AROMA_VEIL`. Would that suffice? I haven't figured out how to display the correct battler ability yet though. 
2. On showdown the ability pop up does show up. 
3. Correct. Should have been obvious 🫣